streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableAggregateTest.java

```diff
@@ -32,38 +34,55 @@
 import org.apache.kafka.streams.kstream.Serialized;
 import org.apache.kafka.streams.kstream.ValueJoiner;
 import org.apache.kafka.streams.kstream.ValueMapper;
+import org.apache.kafka.streams.test.ConsumerRecordFactory;
 import org.apache.kafka.test.KStreamTestDriver;
 import org.apache.kafka.test.MockAggregator;
 import org.apache.kafka.test.MockInitializer;
 import org.apache.kafka.test.MockMapper;
 import org.apache.kafka.test.MockProcessorSupplier;
 import org.apache.kafka.test.TestUtils;
+import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 
-import java.io.File;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Properties;
 
 import static org.junit.Assert.assertEquals;
 
 public class KTableAggregateTest {
 
     final private Serde<String> stringSerde = Serdes.String();
     private final Consumed<String, String> consumed = Consumed.with(stringSerde, stringSerde);
-    private final Serialized<String, String> stringSerialzied = Serialized.with(stringSerde, stringSerde);
+    private final Serialized<String, String> stringSerialized = Serialized.with(stringSerde, stringSerde);
+    private final ConsumerRecordFactory<String, String> recordFactory = new ConsumerRecordFactory<>(new StringSerializer(), new StringSerializer());
+    private TopologyTestDriver driver;
+    private Properties props;
 
-    private File stateDir = null;
-
-    @Rule
-    public EmbeddedKafkaCluster cluster = null;
     @Rule
-    public final KStreamTestDriver driver = new KStreamTestDriver();
+    //TODO: replace this driver with TopologyTestDriver when cache.max.bytes.buffering is working
```

Comment:
Can you explain this?

Reply:
I may be doing something wrong, but setting CACHE_MAX_BYTES_BUFFERING_CONFIG had no effect and I kept getting the same results as I would if the cache was disabled, so tests which rely on caching were failing for `TopologyTestDriver` due to results mismatch.

Reply:
Hmmm. Not entirely sure without digging into details -- maybe the old driver passed this config forward while the new does not (or not correctly) -- might be a bug in `TopologyTestDriver`. Let us know if you need more input on this to figure out the root cause.